extensions/battleroyale/standalone.mjs

```diff
@@ -0,0 +1,519 @@
+          }),
+      );
+    }
+
+    this.cooldown = new Set();
+    this.fireCounter = 0;
+    this.gameStarted = false;
+    this.updateMapIntervalIDs = [];
+    this.participatePlayerIDs = new Set(); // store playerID
+    this.playerOldPosition = new Map(); // [string, mapCoord] store player's position before tp to game
+
+    await this.helper.gameState.registerOnPlayerUpdate((msg) => {
+      try {
+        this.onPlayerUpdate(msg);
+      } catch (e) {
+        console.error('battleroyale register player update listener failed: ', e, e.stack);
+      }
+    });
+
+    // register state functions for NPC
+    await this.helper.callS2sAPI('npc', 'registerStateFunc');
+    const initTime = new Date();
+    const startTime = new Date(Math.floor((+initTime + START_GAME_INTERVAL - 1) / START_GAME_INTERVAL) * START_GAME_INTERVAL + START_GAME_INTERVAL/2);
+    alarm(startTime, this.setGameInterval.bind(this));
+
+    // get config from default
+    if (config.has('battleroyale.arena')) {
+      this.arenaPos = config.get('battleroyale.arena');
+    } else {
+      // default position
+      this.arenaPos = {x: 25, y: 25};
+    }
+  }
+
+  /**
+   * Game interval
+   */
+  setGameInterval() {
+    alarm.recurring(START_GAME_INTERVAL, this.startGame.bind(this));
+  }
+
+  /**
+   * start Game
+   * @return {Boolean} success or not
+   */
+  async startGame() {
+    if (this.gameStarted) {
+      await this.resetGame();
+    }
+    if (this.participatePlayerIDs.size < 3) {
+      for (const playerID of this.participatePlayerIDs) {
+        await this.helper.callS2cAPI(playerID, 'notification', 'showNotification', 3000, '[battleroyale] player not enough to start (less than 3)');
+      }
+      return false;
+    }
+    for (const playerID of this.participatePlayerIDs) {
+      const playerloc = await this.helper.getPlayerLocation(playerID);
+      this.playerOldPosition.set(playerID, playerloc);
+      await this.helper.callS2cAPI(playerID, 'notification', 'showNotification', 3000, '[battleroyale] Game Start!');
+      await this.teleportPlayer(playerID);
+    }
+
+    this.gameStarted = true;
+    const updateMap = setInterval(async () => {
+      for (const mapName of this.arenaOfMaps.keys()) {
+        const needUpdate = this.updateEdge(
+            this.fires, this.arenaOfMaps.get(mapName), this.fireCounter,
+        );
+        if (!needUpdate) {
+          clearInterval(updateMap);
+          return;
+        }
+        await this.helper.broadcastCellSetUpdateToAllUser(
+            'update',
+            mapName,
+            CellSet.fromObject({
+              name: LAYER_FIRE.layerName,
+              cells: Array.from(this.fires),
+            }),
+        );
+        const players = this.helper.gameState.getPlayers();
+        players.forEach(async (player, playerID) => {
+          if (this.helper.gameMap.getCell(player.mapCoord, LAYER_FIRE.layerName)) {
+            await this.killPlayer(playerID);
+          }
+        });
```

Comment:
Just a kind reminder that this piece of code has the same effect as
```javascript
players.forEach((player, playerID) => {
  if (...) this.killPlayer(playerID);
}
```

, since the two versions both create all the promises and not caring about their execution order.

If you want to ensure the execution order, you have to use for loop like this:

```javascript
for (const [playerID, player] of players) {
  if (...) this.killPlayer(playerID);
```

Reply:
The same issue happens on line 254 ~ 259.